### ISSUE TITLE
Item quantity is getting doubled with calling game.itempiles.API.transferItems(

### ISSUE THREAD

**Issue by matthewbstroud** [fantasycalendar/FoundryVTT-ItemPiles#677]
**I confirm the following:**
- [x ] I am on the latest version of Item Piles
- [x] I have tried to clear my cache (Ctrl / Cmd + F5)
- [x] I have checked the console for errors (F12 to open, Console tab)
- [x] I have tried to reset Item Piles' system settings
- [x] I have reproduced this bug with only item piles & its dependency modules enabled

**Describe the bug**
If I have quantity 4 of an item and transfer to an item pile, the quantity ends up being 8 on the vault.

**To Reproduce**

```js
        let result = await game.itempiles.API.createItemPile(pileOptions);
        let backpackToken = await fromUuid(result.tokenUuid);
        let items = actor.items.filter(i => i?.system?.container == backpack.id);
        items.unshift(backpack);
        let transferred = await game.itempiles.API.transferItems(controlledToken, backpackToken, items);
```

**Expected behavior**
I would expect the amounts to be retained.

**Setup**
- Foundry Version: v12 331
- System Version: DND5e 3.3.1
- Item Piles Version: 3.1.6
- Browser & version Edge 130.0.2849.56 (Official build) (64-bit)

**Active modules**
- Item Piles
- libwrapper
- socketlib
- CPR
- Stroud's DnD Helpers

**Comment by matthewbstroud** [comment]
![image](https://github.com/user-attachments/assets/e8e7f100-3d1f-40a6-97ea-16810752d641)
![image](https://github.com/user-attachments/assets/a6d1f3e7-04a0-4ec6-b73e-d931c753ceb2)

**Comment by matthewbstroud** [comment]
appears to be happening around here...
![image](https://github.com/user-attachments/assets/71797557-26bd-4b95-b2ba-016405b631e4)

**Comment by matthewbstroud** [comment]
going into line 300 the quantities look okay.

**Comment by matthewbstroud** [comment]
![image](https://github.com/user-attachments/assets/c1d61ec9-090a-4493-b66f-5592348cb850)

**Comment by matthewbstroud** [comment]
so, there was 1 tinderbox and 1 container, sourceTransaction.prepare() ends up with 2 items to delete and 2 itemDeltas, the delta for the tinderbox should be -1, but it is -2.

**Comment by matthewbstroud** [comment]
the item id appears to be duplicated in the list
![image](https://github.com/user-attachments/assets/759d69b6-c9da-4091-bfcf-321f2a56a672)

**Comment by matthewbstroud** [comment]
The extra item appears to be added here...
![image](https://github.com/user-attachments/assets/5f0a5a1d-8a59-4137-8e76-c9a0e7b15782)

**Comment by rsadamoresten** [comment]
Can you confirm that the backpack is not in your `items` variable?

**Comment by Haxxer** [comment]
Can you confirm that the backpack is not in your `items` variable?

**Comment by matthewbstroud** [comment]
Okay, I think I know what is going on here.  In v11, when you dropped a backpack, it just dropped the backpack and left all it's contents in your inventory, I wrote code to handle that.   So, it would appear that I no longer need my code.

**Comment by matthewbstroud** [comment]
Yeah, you basically corrected an issue I had been coding around in v11.

**Comment by Haxxer** [comment]
It is still an issue, it shouldn't duplicate ids.

**Comment by matthewbstroud** [comment]
Okay, well, the issue is that I was transferring a container _and_ all of its contents.   You have added a handler for containers, that adds their contents.  This caused the items to be duplicated.

**Comment by matthewbstroud** [comment]
The reason is that you do a concat of items and newitems.  It really needs to be a union.

**Comment by matthewbstroud** [comment]
(if the item already exists, it wouldn't be added again.)

**Comment by matthewbstroud** [comment]
I could fix it if you don't have time.

**Comment by matthewbstroud** [comment]
Just let me know.

**Comment by Haxxer** [comment]
Item Piles is open source, so feel free to create a pull request as you please

**Comment by matthewbstroud** [comment]
https://github.com/fantasycalendar/FoundryVTT-ItemPiles/pull/678
